tests/validator_test.go

```diff
@@ -329,4 +337,611 @@ var _ = Describe("Template validator", func() {
+			if !errors.IsNotFound(err) {
+				Expect(err).ToNot(HaveOccurred(), "Failed to Delete VM")
+			}
+		})
+
+		It("should create VM without template", func() {
+			vm = NewVirtualMachine(vmi)
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("be created from template with minimal rules", func() {
+			template = TemplateWithMinimalRule()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			// TODO: Investigate why this test consistently fails without Eventually
+			Eventually(func() error {
+				return apiClient.Create(ctx, vm)
+			}, shortTimeout).Should(BeNil(), "Failed to create VM")
+		})
+		It("[test_id:5033]: Template with validations, VM without validations", func() {
+			template = TemplateWithRules()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 2, "q35", "128M")
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:2960] Negative test - Create a VM with machine type violation", func() {
+			template = TemplateWithRules()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			// set value unfulfilling validation
+			vmi = addDomainResourcesToVMI(vmi, 2, "test", "128M")
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			Expect(errors.IsInvalid(apiClient.Create(ctx, vm))).To(BeTrue(), "Should match error type because of unfulfilled validations")
+		})
+		It("test with template optional rules unfulfilled", func() {
+			template = TemplateWithRulesOptional()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("test with template cpu count 0 ", func() {
+			template = TemplateWithRules()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
```

Comment:
The original test case for this was different. It checked that the validation fails if the VM resource does not contain the jsonpath that the rule needs to check.
For example this VM spec does not contain CPU informataion:
```
spec:
  domain:
    devices:
      interfaces:
      - name: default
        bridge: {}
    machine:
      type: "q35"
    resources:
      requests:
        memory: 128M
  networks:
  - name: default
    pod: {}
  terminationGracePeriodSeconds: 0
```
So a rule that checks `jsonpath::.spec.domain.cpu.cores` will fail, because the path does not exist.

In this new test case, the VM contains CPU information `vmi.Spec.Domain.CPU` , it is not `nil`. So this new test does not test the same functionality as the old test.

Please set the CPU to `nil`, change the the `Expect()` to expect and error, and rewrite the test's description to mention that one of the jsonpaths needed for a rule does not exist in the VM.